### PR TITLE
#138 table of contents (SE Topics: Static Analysis to UI/UX)

### PR DESCRIPTION
**Pull request by iskandarzulkarnaien**
Adds table of contents (toc) to articles without one and hides them except for mobile and print view as discussed in #138 . 

This PR affects the below chapters under `SE Topics`. Kindly use the below checklist for reviewing.

Items to test for each chapter:
- [x] Web view (Expected: in-article toc absent)
- [x] Mobile view (Expected: in-article toc present and **links clickable**, right nav panel absent)
- [x] Print view (Expected: in-article toc present, left and right nav panels absent)

Chapters:
- [x] Introduction to Static Analysis
- [x] CheckStyle
- [x] PMD
- [x] FindBugs
- [x] ESLint
- [x] Integration Testing
- [x] Writing Testable Code
- [x] Accessibility
- [x] Design Systems
- [x] Introduction to UI/UX
- [x] Refining UX Design Skills

Devtools shortcuts
Firefox:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd>

Chrome:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>

### CODE REVIEW

**Comment by iskandarzulkarnaien** [comment]
@yash-chowdhary , @tiuweehan if you guys have the time could you review? Thanks!

**Review by tiuweehan — APPROVED** [review]
LGTM!

File: contents/staticAnalysis/FindBugs.md [hunk 1]
```diff
@@ -15,6 +15,24 @@
...
+<box id="article-toc">
+
+* [Overview](#overview)
+* [Features](#features)
+* [Examples of Bugs that can be found using FindBugs](#examples-of-bugs-that-can-be-found-using-findbugs)
+  * [Incorrectly overriding methods](#incorrectly-overriding-methods)
+  * [Find hash-equals mismatch](#find-hash-equals-mismatch)
+  * [Return value of method ignored](#return-value-of-method-ignored)
+  * [Null pointer dereference](#null-pointer-dereference)
+* [How to use it](#how-to-use-it)
```
Comment (yash-chowdhary) [inline contents/staticAnalysis/FindBugs.md]: To make sure that there's no difference b/w the ToC and the pageNav, we can add the `?` at the end. There are a few other places where we can do this, and I've marked it with 👆 
```suggestion
* [How to use it?](#how-to-use-it)
```

File: contents/staticAnalysis/PMD.md [hunk 2]
```diff
@@ -15,6 +15,24 @@
 
 Author: [Xiao Pu](https://nus-oss.github.io/cs3281-website/students/AY1617S2/xiaoPu/xiaoPu-Resume.html)
 
+<box id="article-toc">
+
+* [Overview‎](#overview)
+* [Features‎](#features)
+* [Limitation‎](#limitation)
+* [How to use it‎](#how-to-use-it)
```
Comment (yash-chowdhary) [inline contents/staticAnalysis/PMD.md]: 👆

File: contents/staticAnalysis/PMD.md [hunk 3]
```diff
@@ -15,6 +15,24 @@
...
+  * [Download‎](#download)
+  * [Configuration‎](#configuration)
+  * [Suppress Warnings‎](#suppress-warnings)
+  * [Command Line Usage‎](#command-line-usage)
+  * [Integration with Build Automation Tools‎](#integration-with-build-automation-tools)
+  * [Integration with IDEs‎](#integration-with-ides)
+* [Available Rulesets‎](#available-rulesets)
+* [How does it work‎](#how-does-it-work)
```
Comment (yash-chowdhary) [inline contents/staticAnalysis/PMD.md]: 👆

File: contents/staticAnalysis/checkStyle.md [hunk 4]
```diff
@@ -15,6 +15,21 @@
...
+
+* [Overview‎](#overview)
+* [Features‎](#features)
+* [Limitations‎](#limitations)
+* [How to use it‎](#how-to-use-it)
+  * [Configuration‎](#configuration)
+  * [Suppress Warnings‎](#suppress-warnings)
+  * [Running‎](#running)
+* [Available Configurations‎](#available-configurations)
+* [How does it work‎](#how-does-it-work)
```
Comment (yash-chowdhary) [inline contents/staticAnalysis/checkStyle.md]: 👆

File: contents/staticAnalysis/checkStyle.md [hunk 5]
```diff
@@ -15,6 +15,21 @@
 
 Author: [Xiao Pu](https://nus-oss.github.io/cs3281-website/students/AY1617S2/xiaoPu/xiaoPu-Resume.html)
 
+<box id="article-toc">
+
+* [Overview‎](#overview)
+* [Features‎](#features)
+* [Limitations‎](#limitations)
+* [How to use it‎](#how-to-use-it)
```
Comment (yash-chowdhary) [inline contents/staticAnalysis/checkStyle.md]: 👆

File: contents/uix/design-systems.md [hunk 6]
```diff
@@ -17,6 +17,17 @@
 
 Reviewers: [Ronak Lakhotia](https://github.com/RonakLakhotia), [Metta Ong](https://github.com/ongspxm)
 
+<box id="article-toc">
+
+* [What is a Design System?](#what-is-a-design-system)
+* [Why should we use a Design System?](#why-should-we-use-a-design-system)
+* [Why not front-end frameworks or style guides?](#why-not-front-end-frameworks-or-style-guides)
+* [Benefits of a Design System](#benefits-of-a-design-system)
+* [How to build a Design System](#how-to-build-a-design-system)
```
Comment (yash-chowdhary) [inline contents/uix/design-systems.md]: 👆

**Review by yash-chowdhary — COMMENTED** [review]
Just a few nits, but LGTM overall! 🚀

**Comment by iskandarzulkarnaien** [comment]
@yash-chowdhary Thanks for spotting those! Have added the question marks in. Could you re-review? Thanks!

**Review by yash-chowdhary — APPROVED** [review]

**Comment by iskandarzulkarnaien** [comment]
Thanks for the help @tiuweehan , @yash-chowdhary !

@damithc this PR is ready to be merged

**Pull request merged**